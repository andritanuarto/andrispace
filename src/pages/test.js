import React, { useState } from 'react';
import Child from './test/child';

const Test = () => {
  const [count, setCount] = useState(0);

  console.log('rerender');
  console.log(count, 'count');

  const a = {a: 'a'};
  const b = a;
  const c = count <= 2 ? a : b;

  const z = 'a';

  return (
    <div>
      <Child ok={a} />
      <button onClick={() => setCount(count + 1)}>Test</button>
    </div>
  )
};

export default Test;