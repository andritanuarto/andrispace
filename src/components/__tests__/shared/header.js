import React from "react";
import Header from "../../shared/header";
import { findByTestAttr } from '../../../test/testUtils';
import Enzyme, { mount } from 'enzyme';
import { Provider } from "react-redux"
import configureStore from 'redux-mock-store';

import EnzymeAdapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new EnzymeAdapter() });

const mockStore = configureStore();
const store = mockStore();

const defaultProps = {
  navColor: '#ffffff',
  logoColor: '#ffffff',
  navigationOpen: '#ffffff',
  initialRender: '#ffffff',
  handleNavigation: '#ffffff',
  handleInitialRenderStatus: '#ffffff'
}

describe('My Footer Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
          <Header {...defaultProps} />
      </Provider>
    );
  });

  it('It should render <Header> properly', () => {
      expect(wrapper.find(Header).length).toBe(1);
      expect(wrapper.find('.header').length).toBe(1);
  });
});