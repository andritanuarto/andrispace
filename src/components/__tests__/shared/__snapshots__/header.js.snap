// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Footer Component should render correctly with default props and redux store 1`] = `ShallowWrapper {}`;
